src/core/modules/AtmosphereShowMySky.cpp

```diff
@@ -724,12 +730,40 @@ void AtmosphereShowMySky::computeColor(StelCore* core, const double JD, const Pl
 	{
 		lastUsedAltitude_ = location.altitude;
 		probeZenithLuminances(location.altitude);
+		dynResTimer=0;
 	}
 
 	auto sunPos  =  sun.getAltAzPosAuto(core);
 	if (std::isnan(sunPos.length()))
 		sunPos.set(0.,0.,-1.*AU);
 
+	if (flagDynamicResolution)
+	{
+		float f1=prj->getFov(), i1=fader.getInterstate();
+		Vec3d p1=sunPos, s1;
+		prj->project(p1,s1);
+		float df=qAbs(prevFov-f1)/(prevFov+f1), di=qAbs(prevFad-i1);
+		double dp=(prevPos-p1).length(), ds=(prevSun-s1).length();
+		dynResTimer++;
+		if (df+di+dp<10e-3 && ds<1 && dynResTimer<0)
+			return;
+
+		ppxatmo=dynResTimer<0?ppxmax:1;
+		if (prevPxa!=ppxatmo)
+			resizeRenderTarget(width, height);
+
+		bool verbose=qApp->property("verbose").toBool();
+		if (verbose)
+			qDebug() << "dynResTimer" << dynResTimer << "ppxatmo" << ppxatmo << "Fov" << df << "Fad" << di << "Pos" << dp << "Sun" << ds;
+
+		dynResTimer=dynResTimer<0?-2:-18;
```

Comment:
These magic numbers need some explanation, like what each of them means, and why they are negative.

Reply:
For example, panning reduces the resolution. when you stop panning, the "-2" causes an early redraw at full resolution.
On the other hand, only every 18th frame is drawn in real-time display (-18).